spec/integration/application/apply_spec.rb

```diff
@@ -577,4 +578,55 @@ def bogus()
       expect(File.read(dest_file)).to eq("content from the module")
     end
   end
+
+  context 'http report processor' do
+    before :all do
+      WebMock.disable!
+    end
+
+    after :all do
+      WebMock.enable!
+    end
+
+    before :each do
+      Puppet[:reports] = 'http'
+
+      # make sure we don't take too long
+      Puppet[:http_connect_timeout] = '5s'
+      Puppet[:server] = '127.0.0.1'
+      Puppet[:certname] = '127.0.0.1'
+
+      Puppet[:localcacert] = File.join(PuppetSpec::FIXTURE_DIR, 'ssl', 'ca.pem')
+      Puppet[:hostcrl] = File.join(PuppetSpec::FIXTURE_DIR, 'ssl', 'crl.pem')
+      Puppet[:hostprivkey] = File.join(PuppetSpec::FIXTURE_DIR, 'ssl', '127.0.0.1-key.pem')
+      Puppet[:hostcert] = File.join(PuppetSpec::FIXTURE_DIR, 'ssl', '127.0.0.1.pem')
+
+      facts = Puppet::Node::Facts.new(Puppet[:certname])
+      Puppet::Node::Facts.indirection.save(facts)
+    end
+
+    let(:apply) { Puppet::Application[:apply] }
+
+    it 'submits a report via reporturl' do
+      report = nil
+
+      response_proc = -> (req, res) {
+        report = Puppet::Transaction::Report.convert_from(:yaml, req.body)
+      }
+
+      https = PuppetSpec::HTTPSServer.new
+      https.start_server(response_proc: response_proc) do |https_port|
+        Puppet[:reporturl] = "https://127.0.0.1:#{https_port}/reports/upload"
+
+        expect {
+          apply.command_line.args = ['-e', 'notify { "hi": }']
+          apply.run
+        }.to exit_with(0)
+         .and output(/Applied catalog/).to_stdout
+
+        expect(report).to be_a(Puppet::Transaction::Report)
+        expect(report.resource_statuses['Notify[hi]']).to be
```

Comment:
this might be more clear as `not nil` rather than just `be`?

Reply:
Updated